File: backend/src/application/repositories/message.repository.ts
import { Message } from '../../domain/message';

export interface IMessageRepository {
  create(message: Omit<Message, 'id' | 'createdAt'>): Promise<Message>;
  findById(id: string): Promise<Message | null>;
  findByChannelId(channelId: string): Promise<Message[]>;
}

export const IMessageRepository = Symbol.for('IMessageRepository');

File: backend/application/repositories/message_repository.go
package repositories

import (
	"backend/domain"
	"context"
)

type MessageRepository interface {
	Create(ctx context.Context, message *domain.Message) (*domain.Message, error)
	FindByID(ctx context.Context, id string) (*domain.Message, error)
	FindByChannelID(ctx context.Context, channelID string) ([]*domain.Message, error)
}
